File: uikit/src/main/java/utils/FontUtils.java
package utils;

import android.content.Context;
import android.graphics.Typeface;

import java.lang.ref.WeakReference;

import javax.inject.Singleton;

public class FontUtils {

    private static FontUtils _instance;

    public static final String robotoMedium = "Roboto-Medium.ttf";

    public static final String robotoBlack = "Roboto-Regular.ttf";

    public static final String robotoRegular = "Roboto-Regular.ttf";

    public static final String robotoBold = "Roboto-Bold.ttf";

    public static final String robotoLight = "Roboto-Light.ttf";

    public static final String robotoThin = "Roboto-Thin.ttf";

    private Context context;

    public static FontUtils getInstance(Context context) {
        if (_instance == null) {
            _instance = new FontUtils(context);
        }
        return _instance;
    }


    private FontUtils(Context context) {
        this.context = context;
    }

    public Typeface getTypeFace(String fontName) {
        Typeface typeface = null;
        if (context != null)
            typeface = Typeface.createFromAsset(context.getAssets(), fontName);
        return typeface;
    }

}

File: uikit-kotlin/src/main/java/com/cometchat/pro/uikit/ui_resources/utils/FontUtils.kt
package com.cometchat.pro.uikit.ui_resources.utils

import android.content.Context
import android.graphics.Typeface

public class FontUtils {


    private var context: Context? = null
    companion object {
        private var _instance: FontUtils? = null

        val robotoMedium = "Roboto-Medium.ttf"

        val robotoBlack = "Roboto-Regular.ttf"

        val robotoRegular = "Roboto-Regular.ttf"

        val robotoBold = "Roboto-Bold.ttf"

        val robotoLight = "Roboto-Light.ttf"

        val robotoThin = "Roboto-Thin.ttf"

//         var context: Context? = null

        fun getInstance(context: Context?): FontUtils {
            if (_instance == null) {
                _instance = FontUtils(context!!)
            }
            return _instance!!
        }



    }
    fun getTypeFace(fontName: String?): Typeface? {
        var typeface: Typeface? = null
        if (context != null) typeface = Typeface.createFromAsset(context!!.assets, fontName)
        return typeface
    }
    constructor(context: Context) {
        this.context = context
    }
}